pkg/cvo/upgradeable.go

```diff
@@ -460,7 +460,14 @@ func (optr *Operator) updateFunc(oldObj, newObj interface{}) {
 }
 
 func (optr *Operator) deleteFunc(obj interface{}) {
-	cm := obj.(*corev1.ConfigMap)
+	if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+		obj = tombstone
+	}
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		klog.Errorf("Unexpected type %T", obj)
+		return
+	}
```

Comment:
This looks good to me. 

I have looked into our other `DeleteFunc` handler and it seems like none of them need a similar treatment because they do not use the object from the informer but just queue something else for processing.

- https://github.com/openshift/cluster-version-operator/blob/9c0a4acf1afd6bd282ab451d2ebcd68c6dbc37f1/pkg/autoupdate/autoupdate.go#L119
- https://github.com/openshift/cluster-version-operator/blob/9c0a4acf1afd6bd282ab451d2ebcd68c6dbc37f1/pkg/cvo/cvo.go#L504-L506
- https://github.com/openshift/cluster-version-operator/blob/9c0a4acf1afd6bd282ab451d2ebcd68c6dbc37f1/pkg/featurechangestopper/featurechangestopper.go#L51-L53

Reply:
And just added 2763543246681d6a4d0a38afa68f2b771314af9c to make that more obvious.